modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/service/impl/SXPBlueprintLocalServiceImpl.java

```diff
@@ -215,7 +216,10 @@ private void _validateExternalReferenceCode(
 			externalReferenceCode, companyId);
 
 		if (sxpBlueprint != null) {
-			throw new DuplicateSXPBlueprintExternalReferenceCodeException();
+			throw new DuplicateSXPBlueprintExternalReferenceCodeException(
+				StringBundler.concat(
+					"Duplicate blueprint external reference code ",
```

Comment:
I think it's not necessary to mention Blueprint (uppercase) here any more as it becomes obvious in the exception itself. Maybe just: "Duplicate external reference code " ...

Reply:
@peerkar I got the pattern from
* CPMeasurementUnitLocalServiceImpl
* JournalFolderLocalServiceImpl
* KBArticleLocalServiceImpl

etc. I thought that this might be a pattern BChan enforces.

Reply:
Then maybe SXPing and uppercasing: SXPBlueprint, SXPElement?

Reply:
If I look at for example KB, I see that the error message is _Duplicate knowledge base article external reference code _. So it's probably _sxp blueprint_, I can change that.

Reply:
I guess that it's meant to be a common noun here: knowledge base, blogs or whatever. If SXPBlueprint is treated as common noun, then I think it should be the way you had it originally, lowercase. 

SXP on the other hand is explicitly a proper noun and lowercasing that would appear strange to me.

Reply:
I see. Let's try to send it as-is then.